common/sched.c

```diff
@@ -138,8 +161,12 @@ task_t *get_task_by_id(tid_t id) {
 task_t *get_task_by_name(const char *name) {
     task_t *task;
 
+    if (!name) {
+        return NULL;
+    }
+
     list_for_each_entry (task, &tasks, list) {
-        if (!strcmp(task->name, name))
+        if (task->name && !strcmp(task->name, name))
```

Comment:
nit/future: at this point it may make sense to add `string_empty()` and `string_equal()` helper routines